src/platforms/posix/px4_layer/px4_posix_tasks.cpp

```diff
@@ -65,6 +65,15 @@
 #define PX4_MAX_TASKS 50
 #define SHELL_TASK_ID (PX4_MAX_TASKS+1)
 
+#if defined(__PX4_POSIX_RPI)
+/**
+ * The minimum stack size of 16384 bytes on Debian Jessie for the Raspberry Pi is too small.
+ * We add the minimum amount possible for it to function properly, in this case 4K.
+ */
+#undef PTHREAD_STACK_MIN
+#define PTHREAD_STACK_MIN 16384 + 4096
```

Comment:
Undefining a define is not good style because anyone assuming this is set to the proper OS values will get badly surprised. For example assuming someone uses the define later to access a buffer and is going over its allocated size.

Even more so you're hardcoding an OS constant to a magic number. That's really asking for trouble :).

Could you change to instead just increase the allocation where its happening to be *always* 4K more than asked for? Please do so for any Linux system, not just RPI. I expect that the fact that Linux is putting some stuff on the task / thread stack is causing issues later on.

I will quickly redo and cover other corner cases. Thanks for providing the 4K mark!